zebra-consensus/src/transaction/tests.rs

```diff
@@ -294,6 +294,50 @@ async fn v4_transaction_with_transparent_transfer_is_accepted() {
     assert_eq!(result, Ok(transaction_hash));
 }
 
+/// Test if V4 coinbase transaction is accepted.
+#[tokio::test]
+async fn v4_coinbase_transaction_is_accepted() {
+    let network = Network::Mainnet;
+
+    let canopy_activation_height = NetworkUpgrade::Canopy
+        .activation_height(network)
+        .expect("Canopy activation height is specified");
+
+    let transaction_block_height =
+        (canopy_activation_height + 10).expect("transaction block height is too large");
+
+    // Create a fake transparent coinbase that should succeed
+    let (input, output) = mock_coinbase_transparent_output(transaction_block_height);
+    let known_utxos = HashMap::new();
+
+    // Create a V4 coinbase transaction
+    let transaction = Transaction::V4 {
+        inputs: vec![input],
+        outputs: vec![output],
+        lock_time: LockTime::Height(block::Height(0)),
+        expiry_height: transaction_block_height,
+        joinsplit_data: None,
+        sapling_shielded_data: None,
+    };
+
+    let transaction_hash = transaction.hash();
+
+    let state_service =
+        service_fn(|_| async { unreachable!("State service should not be called") });
+    let script_verifier = script::Verifier::new(state_service);
+    let verifier = Verifier::new(network, script_verifier);
+
+    let result = verifier
+        .oneshot(Request::Block {
+            transaction: Arc::new(transaction),
+            known_utxos: Arc::new(known_utxos),
```

Comment:
Nit: I'd remove the `known_utxos` variable to avoid the impression that there are any known UTXOs in the test.

```suggestion
            known_utxos: Arc::new(HashMap::new()),
```

Reply:
Fixed in 3ee905f8